docs/framework/wcf/extending/understanding-state-changes.md

```diff
@@ -22,12 +22,12 @@ This topic discusses the states and transitions that channels have, the types us
   
  Every <xref:System.ServiceModel.ICommunicationObject> starts out in the Created state. In this state, an application can configure the object by setting its properties. Once an object is in a state other than Created, it is considered immutable.  
   
- ![Channel state transition](./media/channelstatetranitionshighleveldiagram.gif "ChannelStateTranitionsHighLevelDiagram")  
+ ![Dataflow diagram of the channel state transition.](./media/understanding-state-changes/channel-state-tranitions-high-level-diagram.gif "ChannelStateTranitionsHighLevelDiagram")  
 Figure 1. The ICommunicationObject State Machine.  
   
  Windows Communication Foundation (WCF) provides an abstract base class named <xref:System.ServiceModel.Channels.CommunicationObject> that implements <xref:System.ServiceModel.ICommunicationObject> and the channel state machine. The following graphic is a modified state diagram that is specific to <xref:System.ServiceModel.Channels.CommunicationObject>. In addition to the <xref:System.ServiceModel.ICommunicationObject> state machine, it shows the timing when additional <xref:System.ServiceModel.Channels.CommunicationObject> methods are invoked.  
   
- ![State changes](./media/wcfc-wcfchannelsigure5statetransitionsdetailsc.gif "wcfc_WCFChannelsigure5StateTransitionsDetailsc")  
+ ![Dataflow diagram of CommunicationObject implementation state changes.](./media/understanding-state-changes/wcfc-wcf-channel-sigure-5-state-transitions-details-c.gif "wcfc_WCFChannelsigure5StateTransitionsDetailsc")  
```

Comment:
```suggestion
 ![Dataflow diagram of CommunicationObject implementation state changes.](./media/understanding-state-changes/wcfc-wcf-channel-sigure-5-state-transitions-details-c.gif)  
```

Reply:
I'd keep file name with words that would help with seo

so maybe communicationobject-implementation-state-machine here?

Make similar changes to the other file names used here in this article.